### PR TITLE
Add docs on geolocation

### LINKED ISSUE #220: Docs for geoip2 functionality
I'm using `sentry-onpremise` and I cannot find any info about setting up and using geoip2 in sentry. All I have is a fact that it works in `sentry.io` (for example, in user settings - session history), and an error in logs:

```
sentry.utils.geo: settings.GEOIP_PATH_MMDB not configured
```

I've also seen some GitHub issues with questions about geoip - they've all been closed without any proper explanation. There is no information on the Sentry community forum either - just some unanswered questions.

See also:
https://github.com/getsentry/onpremise/issues/310

### PR DESCRIPTION
**Pull request by chadwhitacre**
Goes with https://github.com/getsentry/onpremise/pull/766.

Closes #220.

Don't merge until https://github.com/getsentry/onpremise/pull/766 is merged **and released**, hopefully will be in 20.12.0.

### CODE REVIEW

File: src/docs/self-hosted/geolocation.mdx [hunk 1]
```diff
@@ -0,0 +1,26 @@
+---
+title: 'Self-Hosted Geolocation'
+---
+
+Sentry can use [MaxMind's free GeoLite2 database](https://dev.maxmind.com/geoip/geoip2/geolite2/) to geolocate IP addresses, providing additional context for error events where the end-user's IP address is known, as well as for the session history of users logging into your Sentry installation. Our on-premise offering bundles MaxMind's [`geoipupdate`](https://hub.docker.com/r/maxmindinc/geoipupdate) tool for initially downloading and periodically updating this dataset. To enable this functionality, you should sign up for [a free MaxMind account](https://www.maxmind.com/en/geolite2/signup), and then tell Sentry about your credentials via a [`docker-compose.override.yml`](https://docs.docker.com/compose/extends/#understanding-multiple-compose-files) file, like so:
```
Comment (BYK) [inline src/docs/self-hosted/geolocation.mdx:5]: Nitz:
```suggestion
Sentry can use [MaxMind's free GeoLite2 database](https://dev.maxmind.com/geoip/geoip2/geolite2/) to geolocate IP addresses, providing additional context for error events where the end-user's IP address is known, as well as for the session history of users logging into your Sentry installation. Our self-hosted offering bundles MaxMind's [`geoipupdate`](https://hub.docker.com/r/maxmindinc/geoipupdate) tool for initially downloading and periodically updating this dataset. To enable this functionality, you should [sign up for a free MaxMind account](https://www.maxmind.com/en/geolite2/signup), and then tell Sentry about your credentials via a [`docker-compose.override.yml`](https://docs.docker.com/compose/extends/#understanding-multiple-compose-files) file, like so:
```

Also, re `docker-compose.override.yml`: let's take it slow and encourage these env variables for now instead?

File: src/docs/self-hosted/geolocation.mdx [hunk 2]
```diff
@@ -0,0 +1,26 @@
...
+```yaml
+version: '3.4'
+services:
+  geoipupdate:
+    environment:
+      GEOIPUPDATE_ACCOUNT_ID: 012345
+      GEOIPUPDATE_LICENSE_KEY: foobar
+```
+
+By default, `onpremise` is configured to refresh the GeoLite2 database when the `geoipupdate` service first starts, and weekly thereafter. You can change the update frequency with an additional override:
```
Comment (BYK) [inline src/docs/self-hosted/geolocation.mdx:16]:
```suggestion
By default, self-hosted instances are configured to refresh the GeoLite2 database when the `geoipupdate` service first starts, and weekly thereafter. You can change the update frequency with an additional override:
```

**Review by BYK — CHANGES_REQUESTED** [review]
Looks pretty good overall. Let's split the `docker-compose.override.yml` related things as a follow up and we're good to go.

File: src/docs/self-hosted/geolocation.mdx [hunk 3]
```diff
@@ -0,0 +1,18 @@
...
+export GEOIPUPDATE_ACCOUNT_ID="012345"
+export GEOIPUPDATE_LICENSE_KEY="foobar"
+```
+
+By default, self-hosted instances are configured to refresh the GeoLite2 database when the `geoipupdate` service first starts, and weekly thereafter. You can change the update frequency with an additional override:
+
+```shell
+export GEOIPUPDATE_ACCOUNT_ID="012345"
+export GEOIPUPDATE_LICENSE_KEY="foobar"
+export GEOIPUPDATE_FREQUENCY="24" # hours
```
Comment (BYK) [inline src/docs/self-hosted/geolocation.mdx:17]: For this to work, [this line](https://github.com/getsentry/onpremise/pull/766/files#diff-e45e45baeda1c1e73482975a664062aa56f20c03dd9d64a827aba57775bed0d3R134) should not define the value and the default value should be defined in the `.env` file.

**Review by BYK — APPROVED** [review]
I think we could have kept this PR focused solely on geolocation settings but I think the changes are minor and good enough that I won't push back on that.

I'll approve even though there's an inaccuracy that needs to be addressed in getsentry/onpremise#766

File: src/docs/self-hosted/geolocation.mdx (hunk 3 above)
Comment (chadwhitacre) [reply src/docs/self-hosted/geolocation.mdx:17]: Ah, right. Vestige of `docker-compose.override.yml`. Updated over in the `onpremise` PR.

**Comment by chadwhitacre** [comment]
I can take out the index.mdx and sidebar changes if you want. I thought it was confusing to talk about configuring with env vars (or `docker-compose.override.yml`) on the geolocation page when on the main page we say everything can be done from `config.yml`andalsosettings.conf.py. I also found myself struggling to discover and then navigate to the self-hosting subpages (the sidebar is pretty busy), so it seemed good to link those in the main body somewhere.

**Comment by chadwhitacre** [comment]
@PeloWriter Mind giving this a once-over? :) Here are the primary things changed:

1. https://develop-git-chadwhitacre-geolocation.sentry.dev/self-hosted/geolocation/ ... new page
1. https://develop-git-chadwhitacre-geolocation.sentry.dev/self-hosted/#configuration vs. https://develop.sentry.dev/self-hosted/#configuration

File: src/docs/self-hosted/geolocation.mdx [hunk 4]
```diff
@@ -0,0 +1,34 @@
+---
+title: 'Self-Hosted Geolocation'
+---
+
+Sentry can use [MaxMind's free `GeoLite2-City` database](https://dev.maxmind.com/geoip/geoip2/geolite2/) to geolocate IP addresses, providing additional context for error events where the end-user's IP address is known, and for the session history of users logging into your Sentry installation. We bundle MaxMind's [`geoipupdate`](https://hub.docker.com/r/maxmindinc/geoipupdate) tool for this purpose. To enable IP address geolocation, [sign up for a free MaxMind account](https://www.maxmind.com/en/geolite2/signup), and then tell Sentry about your credentials by placing your MaxMind configuration file at `geoip/GeoIP.conf`.
```
Comment (PeloWriter) [inline src/docs/self-hosted/geolocation.mdx:5]:
```suggestion
Sentry can use [MaxMind's free `GeoLite2-City` database](https://dev.maxmind.com/geoip/geoip2/geolite2/) to geolocate IP addresses, providing additional context for error events where the end user's IP address is known and for the session history of users logging into your Sentry installation. We bundle MaxMind's [`geoipupdate`](https://hub.docker.com/r/maxmindinc/geoipupdate) tool for this purpose. To enable IP address geolocation, [sign up for a free MaxMind account](https://www.maxmind.com/en/geolite2/signup), then tell Sentry about your credentials by placing your MaxMind configuration file at `geoip/GeoIP.conf`.
```

File: src/docs/self-hosted/geolocation.mdx [hunk 5]
```diff
@@ -0,0 +1,34 @@
...
+
+```
+AccountID 012345
+LicenseKey foobarbazbuz
+EditionIDs GeoLite2-City
+```
+
+With this configuration file in place, subsequent runs of Sentry's `install.sh` will refresh the IP address geolocation database, and the next time you restart your self-hosted Sentry instance (the `relay` and `web` services, in particular), you should start seeing the latest data. Here's how to confirm that it's working:
```
Comment (PeloWriter) [inline src/docs/self-hosted/geolocation.mdx:13]:
```suggestion
With this configuration file in place, subsequent runs of Sentry's `install.sh` will refresh the IP address geolocation database. The next time you restart your self-hosted Sentry instance (the `relay` and `web` services, in particular), you should start seeing the latest data. Here's how to confirm that it's working:
```

File: src/docs/self-hosted/geolocation.mdx [hunk 6]
```diff
@@ -0,0 +1,34 @@
...
+
+1. For the `relay` service: Dashboards > Errors by Country should have some purple on it.
```
Comment (PeloWriter) [inline src/docs/self-hosted/geolocation.mdx:15]: Bold the path to describe navigation:
```suggestion
1. For the `relay` service: **Dashboards > Errors by Country** should have some purple on it.
```

File: src/docs/self-hosted/geolocation.mdx [hunk 7]
```diff
@@ -0,0 +1,34 @@
...
+
+1. For the `web` service: User Settings > Security > Session History should show country code and region (e.g., "US (CA)") underneath the IP addresses in the table.
```
Comment (PeloWriter) [inline src/docs/self-hosted/geolocation.mdx:17]: And please avoid Latin terms
```suggestion
1. For the `web` service: **User Settings > Security > Session History** should display country code and region (for example, "US (CA)") underneath the IP addresses in the table.
```

File: src/docs/self-hosted/index.mdx [hunk 8]
```diff
@@ -10,9 +10,25 @@ Our recommendation is to download the [latest release of the onpremise reposito
 
 ## Configuration
 
-You are very likely to want to change the default configuration for Sentry. Most, if not all, configuration options that you'd like to tweak are in `sentry/config.yml`, which is generated from [`sentry/config.example.yml`](https://github.com/getsentry/onpremise/blob/master/sentry/config.example.yml) at the time of installation. The file itself documents the most common configuration options as code comments, and you can find more about configuring Sentry over at <Link to="/config/">the configuration section of our developer documentation</Link>. You would likely need to change `system.url-prefix` (this is also prompted at the welcome screen, right after the installation) and settings starting with `mail.` even if we ship with a built-in SMTP server for basic stuff. GitHub and Slack integration settings also reside in this file. For more advanced things, you can have a look at [`sentry/sentry.conf.example.py`](https://github.com/getsentry/onpremise/blob/master/sentry/sentry.conf.example.py).
+You are very likely to want to change the default configuration for Sentry. Here are the main facilities available to you for this purpose:
```
Comment (PeloWriter) [inline src/docs/self-hosted/index.mdx:13]:
```suggestion
You very likely will want to adjust the default configuration for Sentry. These main facilities are available for that purpose:
```

File: src/docs/self-hosted/index.mdx [hunk 9]
```diff
@@ -10,9 +10,25 @@ Our recommendation is to download the [latest release of the onpremise reposito
...
 
-**Once you change your configuration, you'll need to restart all Sentry services by running `docker-compose restart web worker cron sentry-cleanup` (or just `docker-compose restart` to restart everything).**
+1. Most, if not all, configuration options that you'd like to tweak are in `sentry/config.yml`, which is generated from [`sentry/config.example.yml`](https://github.com/getsentry/onpremise/blob/master/sentry/config.example.yml) at the time of installation. The file itself documents the most common configuration options as code comments, and you can find more about configuring Sentry over at <Link to="/config/">the configuration section of our developer documentation</Link>. You would likely need to change `system.url-prefix` (this is also prompted at the welcome screen, right after the installation) and settings starting with `mail.` (though we do ship with a basic SMTP server).  GitHub and Slack integration settings also reside in this file.
```
Comment (PeloWriter) [inline src/docs/self-hosted/index.mdx:15]: Keep the steps succinct
```suggestion
1. `sentry/config.yml` which contains most, if not all, configuration options to adjust. This file is generated from [`sentry/config.example.yml`](https://github.com/getsentry/onpremise/blob/master/sentry/config.example.yml) at the time of installation. 

The file itself documents the most common configuration options as code comments. You can find more about configuring Sentry at [the configuration section of our developer documentation](/config/). Typically, you'll need to change `system.url-prefix` (we prompt you to do this at the welcome screen, right after the installation) and settings starting with `mail.` (though we do ship with a basic SMTP server).  GitHub and Slack integration settings also reside in this file.
```

File: src/docs/self-hosted/index.mdx [hunk 10]
```diff
@@ -10,9 +10,25 @@ Our recommendation is to download the [latest release of the onpremise reposito
...
+
+1. For more advanced things, you can have a look at [`sentry/sentry.conf.example.py`](https://github.com/getsentry/onpremise/blob/master/sentry/sentry.conf.example.py).
```
Comment (PeloWriter) [inline src/docs/self-hosted/index.mdx:17]:
```suggestion
1. [`sentry/sentry.conf.example.py`](https://github.com/getsentry/onpremise/blob/master/sentry/sentry.conf.example.py), which contains more advanced configuration.
```

File: src/docs/self-hosted/index.mdx [hunk 11]
```diff
@@ -10,9 +10,25 @@ Our recommendation is to download the [latest release of the onpremise reposito
...
+
+1. There are a few places where we depend on environment variables. The keys are defined in [.env](https://github.com/getsentry/onpremise/blob/master/.env), though if you need to override any of them we expect you to use some other means of setting environment variables dependent on your system.
```
Comment (PeloWriter) [inline src/docs/self-hosted/index.mdx:19]:
```suggestion
1. Environment variables, the keys for which are defined in [.env](https://github.com/getsentry/onpremise/blob/master/.env). If you need to override any of them, we expect you to use some other means of setting environment variables dependent on your system.
```

File: src/docs/self-hosted/index.mdx [hunk 12]
```diff
@@ -10,9 +10,25 @@ Our recommendation is to download the [latest release of the onpremise reposito
...
+
+1. <Link to="/self-hosted/geolocation/">Geolocation</Link> uses a custom configuration file to conform to the underlying technology.
```
Comment (PeloWriter) [inline src/docs/self-hosted/index.mdx:21]:
```suggestion
1. [Geolocation](/self-hosted/geolocation/) uses a custom configuration file to conform to the underlying technology.
```

File: src/docs/self-hosted/index.mdx [hunk 13]
```diff
@@ -10,9 +10,25 @@ Our recommendation is to download the [latest release of the onpremise reposito
...
+
+<Alert title="Note" level="info">
+  Once you change your configuration, you'll need to restart all Sentry services by running <code>docker-compose restart web worker cron sentry-cleanup</code> (or just <code>docker-compose restart</code> to restart everything).
+</Alert>
+
+Here is further information on specific configuration topics:
+
+- <Link to="/self-hosted/email/">Email</Link>
+- <Link to="/self-hosted/geolocation/">Geolocation</Link>
+- <Link to="/self-hosted/sso/">Single Sign-On (SSO)</Link>
```
Comment (PeloWriter) [inline src/docs/self-hosted/index.mdx:31]: You can use Markdown for these links, from our Tips to Contributors, here's how we link link to another page in docs:
1.  Put the linking text inside brackets [ ]
2. After the brackets, put the file path inside ()

For example:
[PagerDuty](/product/integrations/pagerduty/)

**Review by PeloWriter — APPROVED** [review]
Some fine tuning, otherwise LGTM.

File: src/docs/self-hosted/index.mdx (hunk 13 above)
Comment (chadwhitacre) [reply src/docs/self-hosted/index.mdx:31]: Ah! I used Markdown syntax initially, but I had a relative URL (`./geolocation`), and gatsby was erroring out. I've confirmed that it works with an absolute URL. 👍

**Comment by chadwhitacre** [comment]
Thanks for the review @PeloWriter! I've incorporated your suggestions into the Geolocation doc. For the overview, I agree with your changes but there are some issues to work through, mainly the effect on the list of splitting a new paragraph under bullet 1:

Screen Shot 2020-12-11 at 1 28 15 PM

@BYK was already suggesting we hold the config overview changes for later, so I've gone ahead and done that now. Will pick up with that section in a new PR.

**Comment by PeloWriter** [comment]
Yes, formatting can be tricky with indentation. Let me know if you need help when you pick this up again.

**Pull request merged**